src/scatter.ts

```diff
@@ -0,0 +1,169 @@
+    .attr("width", width + margin.left + margin.right)
+    .attr("height", height + margin.top + margin.bottom)
+    .append("g")
+    .attr("transform",
+        "translate(" + margin.left + "," + margin.top + ")");
+
+    d3.select("#dropdown_container_x").selectAll("select").remove();
+    d3.select("#dropdown_container_y").selectAll("select").remove();
+
+    // Clickable options in the dropdown list
+    const dropdown_list = [
+        "weapon damage", "weapon max range", "movement speed", 
+        "gold costs", "attack duration", "armor",
+        "modifier max value"
+    ];
+    
+    createDropDown(dropdown_list, "x_selection", "x");
+    createDropDown(dropdown_list.reverse(), "y_selection", "y");
+    
+    d3.select('#x_selection')
+    .on('change', function() {
+        updateScatterPlot(data, width, height, margin);       
+    });
+
+    d3.select('#y_selection')
+    .on('change', function() {
+        updateScatterPlot(data, width, height, margin);
+    });
+
+    updateScatterPlot(data, width, height, margin);
+
+    return svg.node();
+}
+
+/**
+ * Returns specific data to plot based on the selected option for a specific axis
+ * @param  {Unit}   unit           unit to return data from
+ * @param  {string} selection_name option selected in dropdown list
+ * @return {number}             numerical value to plot in the scatterplot
+ */
+function returnData(unit: Unit, selection_name: string, debug = false): number {
+
+    switch(selection_name) {
+        case "weapon damage" : return unit.weapons.map(wp => wp.damage)[0];
+        case "weapon max range" : return unit.weapons.map(wp => wp.range.max)[0];
+        case "movement speed" : return unit.movement.speed;
+        case "gold costs" : return unit.costs.gold;
+        case "modifier max value" : {
+            if (debug) console.log("WEAPONS: " + JSON.stringify(unit.weapons.map(wp => wp.modifiers)));
+
+            return unit.weapons.map(wp => {
+                const modifierValues = wp.modifiers.map(m => m.value === null ? 0 : m.value);
+                return modifierValues[0];
+            })[0];
+        };
+        case "attack duration" : return unit.weapons.map(wp => wp.durations.attack)[0];
+        case "armor" : return unit.armor.map(arm => arm.value)[0]; // check if there are more types of armor
+    }
+}
+
+/**
+ * Make a dropdown list for a given axis.
+ */
+function createDropDown(data: string[], name: string, axisName: string) {
+    var dropDown = d3.select("#dropdown_container_" + axisName)
+        .append("select")
+        .attr("class", "selection")
+        .attr("id", name);
+
+    var options = dropDown.selectAll("option")
+        .data(data)
+        .enter()
+        .append("option");
+        options.text(function(d) {
+            return d;
+        });
+}
+
+/**
+ * Update the plot to show the data corresponding to the dropdown selection. 
+ */
+function updateScatterPlot(data: Unit[], width: any, height: any, margin: any) {
```

Comment:
Why is it necessary to separate scatter plot creation and update? It seems as though there is a significant amount of code which is identical.

Reply:
It needs to be updated based on the axes the user picks from the dropdown lists. So every time the user changes the attributes shown on an axis the data should be updated as well

Reply:
I did not think there were many identical elements, but let me know if you have suggestions and I will improve it

Reply:
For example, if you use D3's `join` function instead of `append`, then D3 will automatically add and remove elements as needed, and you can create the elements every time instead of having separate create and update functions. This is how I've seen it done in other repositories and will indeed result in less repetition (the first ~20 lines are the same in both functions). I also have to regenerate the my table based on potentially complex search terms and filters and have successfully employed this approach.

Reply:
I removed the duplicate code, we will add some more improvements once all the graphs are merged together